server/controllers/profile.js

```diff
@@ -0,0 +1,82 @@
+const Profile = require("../models/Profile");
+const asyncHandler = require("express-async-handler");
+
+exports.profileCreatePost = asyncHandler(async (req, res) => {
+  const {
+    firstName,
+    lastName,
+    description,
+    availability,
+    email,
+    username,
+    gender,
+    birthday,
+    phone,
+    address,
+    available,
+  } = req.body;
+  const id = req.user.id;
+
+  if (!firstName || !lastName || !id) {
+    res.status(400);
+    throw new Error("Invalid name");
+  }
+  const profile = await Profile.create({
+    userId: id,
+    firstName,
+    lastName,
+    description,
+    availability,
+    email,
+    username,
+    gender,
+    birthday,
+    phone,
+    address,
+    available,
+  });
+  res.status(201).json({ profile });
+});
+
+exports.profileUpdatePost = asyncHandler(async (req, res) => {
+  const { firstName, lastName, description, availability, email } = req.body;
+  const id = req.user.id;
+  const idExists = await Profile.findOne({ _id: id });
```

Comment:
```suggestion
  const idExists = await Profile.findOne({ userId: id });
```
Won't we need to search for the 'userId' field as it links a profile with a user?